zanata-war/src/main/java/org/zanata/dao/TextFlowTargetHistoryDAO.java

```diff
@@ -75,54 +84,109 @@ public TextFlowTargetHistoryDAO(Session session) {
      * @return list of Object[wordCount][contentState][localeId]
      */
     @NativeQuery
-    public List<Object[]> getUserContributionStatisticInVersion(
+    public List<Object[]> getUserTranslationStatisticInVersion(
             Long versionId, Long personId, Date fromDate, Date toDate, boolean automatedEntry) {
+        Query query = buildContributionStatisticQuery(true, versionId,
+            personId, fromDate, toDate, automatedEntry);
+        query.setComment("textFlowTargetHistoryDAO.getUserTranslationStatisticInVersion");
+        return query.list();
+    }
+
+    /**
+     * Query to get total wordCount of a person(reviewed_by_id) from
+     * HTextFlowTarget union HTextFlowTargetHistory tables
+     * in a project-version, within given date range group by state and locale.
+     *
+     * HTextFlowTargetHistory:
+     * gets latest of all records reviewed from user in given version,
+     * locale and dateRange and its target is not translated by same person.
+     *
+     * HTextFlowTarget:
+     * gets all records reviewed from user in given version, locale and
+     * dateRange.
+     *
+     * @param versionId HProjectIteration identifier
+     * @param personId HPerson identifier
+     * @param from start of date range
+     * @param to end of date range
+     *
+     * @return list of Object[wordCount][contentState][localeId]
+     */
+    @NativeQuery
+    public List<Object[]> getUserReviewStatisticInVersion(
+            Long versionId, Long personId, Date fromDate, Date toDate,
+            boolean automatedEntry) {
+        Query query = buildContributionStatisticQuery(false, versionId,
+                personId, fromDate, toDate, automatedEntry);
+        query.setComment("textFlowTargetHistoryDAO.getUserReviewStatisticInVersion");
+        return query.list();
+    }
+
+    private Query buildContributionStatisticQuery(boolean translations,
+            Long versionId, Long personId, Date fromDate, Date toDate,
+            boolean automatedEntry) {
+        String lastModifiedColumn =
+                translations ? "translated_by_id" : "reviewed_by_id";
```

Comment:
why not just pass in the string value as parameter instead of using a boolean. It will be easier to read too